app/controllers/application_controller.rb

```diff
@@ -23,12 +28,23 @@ class ApplicationController < ActionController::Base
     end
   end
 
-  rescue_from EditionAssertions::AccessError do |e|
-    Rails.logger.warn(e.message)
-    render "documents/forbidden", status: :forbidden, assigns: { edition: e.edition }
-  end
-
   def rendering_context
     request.headers["Content-Publisher-Rendering-Context"] || "application"
   end
+
+  def check_user_access
+    document_id = request.path_parameters[:document]
```

Comment:
nice find 👍 minor, but I'd probably call this document_param than document_id since that'd imply an integer whereas this is what you get when you run to_param on a document.

Reply:
Also forgot to do this earlier - fixed.